src/Routing/UrlBuilder.php

```diff
@@ -83,16 +83,23 @@ private function routeData()
 
     private function slugify($value)
     {
-        $underscorePlaceholder = strtolower(str_random(16));
-        $slashPlaceholder = strtolower(str_random(16));
+        $slashPlaceholder = strtolower(str_random());
 
-        $value = str_replace('_', $underscorePlaceholder, $value);
         $value = str_replace('/', $slashPlaceholder, $value);
 
+        $hasDot = str_contains($value, '.');
+        if ($hasDot) {
+            $dotPlaceholder = strtolower(str_random());
+            $extension = '.'.Str::after($value, '.');
+            $value = str_replace($extension, $dotPlaceholder, $value);
+        }
```

Comment:
I think you are still hung up on the file extension.

Replace dots with the placeholder before, and replace the placeholder with dots after.

Reply:
Got it.